app/controllers/samples_controller.rb

```diff
@@ -22,6 +22,17 @@ def index
       .map { |uuid, name| { value: uuid, label: name } }
 
     @samples = perform_pagination(@samples)
+      .preload(:batch, :sample_identifiers)
```

Comment:
Note that I use `#preload` that generates an explicit SQL request for each association to preload, instead of `#includes` that would often SELECT and JOIN right into the initial SQL request. I always prefer explicit requests, which are much simpler (and saner).